src/nvim/api/win_config.c

```diff
@@ -272,12 +272,12 @@ Window nvim_open_win(Buffer buffer, Boolean enter, Dict(float_config) *config, E
   restore_win_noblock(&switchwin, true);
   if (enter) {
     goto_tabpage_win(tp, wp);
-    if (tabpage_win_valid(tp, wp) && curwin != wp) {
+    if (!tabpage_win_valid(tp, wp) || curwin != wp) {
       api_set_error(err, kErrorTypeException, "Failed to enter tabpage");
       return 0;
     }
   }
-  if (tabpage_win_valid(tp, wp) && buf != wp->w_buffer) {
+  if (buf != wp->w_buffer) {
     win_set_buf(wp, buf, !enter || fconfig.noautocmd, err);
   }
   if (!tabpage_win_valid(tp, wp)) {
```

Comment:
Actually autocommands may even close `tp` and move the window to another tab, so using  `tabpage_win_valid(tp, wp)` is not safe. I think it should be like this:
- Remove this error message as it wasn't previously present.
- Use `win_valid_any_tab(wp)` in both of the two `if` statements below.

Reply:
Hmm alright, that makes sense. Should we still check for curwin on enter then? (assuming the autocmds can change the current window, which afaik they can)

Reply:
Previously there was no check, so I think no.

Reply:
Sounds good